specs/psl-sql-conversion.md

```diff
@@ -0,0 +1,459 @@
+    id Int @id
+    enum TestEnum
+}
+
+enum TestEnum{
+  A
+  B
+}
+```
+
+We handle the validation of Enum values in the core, and place no constraints on the db column.  
+
+```sql
+CREATE TABLE "test"."User" (
+  "enum" text NOT NULL DEFAULT 'A' ,
+  "id" SERIAL,
+  PRIMARY KEY ("id")
+);
+```
+
+🚨 We do not use any DB capabilities for enums, therefore when introspecting again, this is just a String field for us. 
+
+
+
+## Mapping Default Values 
+
+```groovy
+model User {
+  id       Int       @id
+  string   String    @default("test")
+  date     DateTime  @default(now())
+}
+```
+
+```sql
+CREATE TABLE "test"."User" (
+  "date" timestamp(3) NOT NULL DEFAULT '1970-01-01 00:00:00' ,
+  "id" SERIAL,
+  "string" text NOT NULL DEFAULT 'test' ,
+  PRIMARY KEY ("id")
+);
+```
+
+🚨 DateTime default values don't yet work. The `now()` expression should translate to something like `CURRENT_TIMESTAMP`. 
+
+
+
+
+## Mapping Relations
+
+### Mapping a one to one relation
+
+There are four potential cases for 1to1 relations when it comes to fields being required. But two of them are always resulting in the same SQL structure. 
+
+Both sides required:
+```groovy
+model User {
+    id      String @id @default(cuid())
+    address Address @relation(references: [id])
+}
+model Address {
+    id   String @id @default(cuid())
+    user User
+}
+```
+
+One side required:
+```groovy
+model User {
+    id      String @id @default(cuid())
+    address? Address @relation(references: [id])
+}
+model Address {
+    id   String @id @default(cuid())
+    user User
+}
+```
+
+Resulting SQL Example (Postgres):
+```sql
+CREATE TABLE "test"."User" (
+  "id" text NOT NULL  ,
+  PRIMARY KEY ("id")
+);
+
+CREATE TABLE "test"."Address" (
+  "id" text NOT NULL  ,
+  PRIMARY KEY ("id")
+);
+
+ALTER TABLE "test"."User" ADD COLUMN "address" text NOT NULL   REFERENCES "test"."Address"("id") ON DELETE SET NULL;
+```
+
+Other side required:
+```groovy
+model User {
+    id      String @id @default(cuid())
+    address Address @relation(references: [id])
+}
+model Address {
+    id   String @id @default(cuid())
+    user User?
+}
+```
+
+Neither side required:
+```groovy
+model User {
+    id      String @id @default(cuid())
+    address? Address @relation(references: [id])
+}
+model Address {
+    id   String @id @default(cuid())
+    user User?
+}
+```
+
+Resulting SQL Example (Postgres):
+```sql
+CREATE TABLE "test"."User" (
+  "id" text NOT NULL  ,
+  PRIMARY KEY ("id")
+);
+
+CREATE TABLE "test"."Address" (
+  "id" text NOT NULL  ,
+  PRIMARY KEY ("id")
+);
+
+ALTER TABLE "test"."User" ADD COLUMN "address" text   REFERENCES "test"."Address"("id") ON DELETE SET NULL;
+```
+
+For a one to one relation the user must specify the side where the foreign key gets stored by annotating the field with `@relation(references: [id])`. This annotated field is mapped to a column with a foreign key constraint. This foreign key constraint is set to null if the referenced row is deleted.
```

Comment:
I'm not sure what the current logic is in the migration engine regarding 1:1 relations, I'll make myself a todo to look it up.